geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/GenerateFeatureWrappers.scala

```diff
@@ -0,0 +1,176 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.utils.geotools
+
+import java.io.{File, FileWriter}
+
+import com.typesafe.config.ConfigFactory
+import org.opengis.feature.`type`.AttributeDescriptor
+import org.opengis.feature.simple.SimpleFeatureType
+
+
+case class AttributeDetails(name: String, index: Int, clazz: String) {
+  def getter: String = s"def $name(): $clazz = sf.getAttribute($index).asInstanceOf[$clazz]"
+  def optionGetter: String = s"def ${name}Opt(): Option[$clazz] = Option($name())"
+  def setter: String = s"def set${name.capitalize}(x: $clazz): Unit = sf.setAttribute($index, x)"
+}
+
+object AttributeDetails {
+  import org.locationtech.geomesa.utils.geotools.Conversions.RichAttributeDescriptor
+
+  def apply(ad: AttributeDescriptor, sft: SimpleFeatureType): AttributeDetails = {
+    val majorBinding = classToString(Some(ad.getType.getBinding))
+    val binding = if (ad.isCollection) {
+      val subtype = classToString(SimpleFeatureTypes.getCollectionType(ad))
+      s"$majorBinding[$subtype]"
+    } else if (ad.isMap) {
+      val keyType = classToString(Option(ad.getUserData.get("keyclass").asInstanceOf[Class[_]]))
+      val valueType = classToString(Option(ad.getUserData.get("valueclass").asInstanceOf[Class[_]]))
+      s"$majorBinding[$keyType,$valueType]"
+    } else {
+      majorBinding
+    }
+    AttributeDetails(ad.getLocalName, sft.indexOf(ad.getLocalName), binding)
+  }
+
+  private def classToString(clas: Option[Class[_]]) = clas.map(_.getCanonicalName).getOrElse("String")
+}
+
+object GenerateFeatureWrappers {
+
+  val className = "SimpleFeatureWrappers"
+
+  /**
+   * Builds all implicit classes in a wrapper object
+   *
+   * @param sfts
+   * @param pkg
+   * @return
+   */
+  def buildAllClasses(sfts: Seq[SimpleFeatureType], pkg: String): String = {
+    val sb = new StringBuilder()
+    sb.append(s"package $pkg\n\n")
+    sb.append(s"object $className {")
+    sfts.foreach(sft => sb.append("\n\n").append(buildClass(sft, "  ")))
+    sb.append("\n}")
+    sb.toString()
+  }
+
+  /**
+   * Builds a single implicit class
+   *
+   * @param sft
+   * @param tab
+   * @return
+   */
+  def buildClass(sft: SimpleFeatureType, tab: String): String = {
+    import scala.collection.JavaConversions._
+
+    val attrs = sft.getAttributeDescriptors.map(AttributeDetails(_, sft))
+
+    val sb = new StringBuilder()
+    sb.append(s"${tab}implicit class ${sft.getTypeName}")
+    sb.append("(val sf: org.opengis.feature.simple.SimpleFeature) extends AnyVal {\n")
+    attrs.foreach { a =>
+      sb.append("\n")
+      sb.append(s"$tab  ${a.getter}\n")
```

Comment:
I changed it so we wouldn't have to loop 3 times. If it bothers you though, I can change back.